functions/system.bash

```diff
@@ -78,49 +70,51 @@ locale_setting() {
     echo "$(timestamp) [openHABian] Setting locale based on user choice... "
     dpkg-reconfigure locales
     loc=$(grep "LANG=" /etc/default/locale | sed 's/LANG=//g')
-    cond_redirect update-locale LANG=$loc LC_ALL=$loc LC_CTYPE=$loc LANGUAGE=$loc
+    cond_redirect update-locale "LANG=$loc LC_ALL=$loc LC_CTYPE=$loc LANGUAGE=$loc"
     whiptail --title "Change Locale" --msgbox "For the locale change to take effect, please reboot your system now." 10 60
     return 0
   fi
 
   echo -n "$(timestamp) [openHABian] Setting locale based on openhabian.conf... "
+  # shellcheck source=/etc/openhabian.conf disable=SC1091
   source "$CONFIGFILE"
   if is_ubuntu; then
-    cond_redirect locale-gen $locales
+    # shellcheck disable=2154
+    cond_redirect locale-gen "$locales"
   else
     touch /etc/locale.gen
     for loc in $locales; do sed -i "/$loc/s/^# //g" /etc/locale.gen; done
     cond_redirect locale-gen
   fi
   cond_redirect dpkg-reconfigure --frontend=noninteractive locales
-  cond_redirect LANG=$system_default_locale; export LANG &>/dev/null
-  cond_redirect LC_ALL=$system_default_locale; export LC_ALL &>/dev/null
-  cond_redirect LC_CTYPE=$system_default_locale; export LC_CTYPE &>/dev/null
-  cond_redirect LANGUAGE=$system_default_locale; export LANGUAGE &>/dev/null
-  cond_redirect update-locale LANG=$system_default_locale LC_ALL=$system_default_locale LC_CTYPE=$system_default_locale LANGUAGE=$system_default_locale
-  if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; fi
+  cond_redirect "LANG=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_ALL=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_CTYPE=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LANGUAGE=${system_default_locale:-en_US.UTF-8}"
+  export LANG LC_ALL LC_CTYPE LANGUAGE
```

Comment:
You have left out `&>/dev/null` here compared to the "old" export commands. How does that differ, is it better?

Reply:
Hmm, I was hunting a strange error message that only showed up with the LC_ALL variable line

2019-09-03_21:19:04_UTC [openHABian] Setting locale based on openhabian.conf... 
/opt/openhabian/functions/system.bash: line 96: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory

Which all by itself is already pretty weird as that's just a variable assignment but obviously treated somehow special .... I think that's why that to /dev/null was in there.
Now as that line didn't work anyway, I removed that in another of my PRs (hope I didn't forget this when splitting my 'big' PR but if so we'll be seeing that error message again).

Reply:
Good. Let's try this pattern and fix later otherwise. Only a cosmetic issue if this slips through.